pkg/snapstore/utils.go

```diff
@@ -179,36 +178,25 @@ func adaptPrefix(snap *brtypes.Snapshot, snapstorePrefix string) string {
 	return snapstorePrefix
 }
 
-// GetSnapstoreSecretHash returns the hash of object store secrets hash
-func GetSnapstoreSecretHash(config *brtypes.SnapstoreConfig) (string, error) {
+// GetSnapstoreSecretModTime returns the latest modification timestamp of the access credential files.
+// Returns an error if fetching the timestamp of the access credential files fails.
+func GetSnapstoreSecretModTime(config *brtypes.SnapstoreConfig) (time.Time, error) {
 	switch config.Provider {
 	case brtypes.SnapstoreProviderLocal:
-		return "", nil
+		return time.Time{}, nil
 	case brtypes.SnapstoreProviderS3:
-		return S3SnapStoreHash(config)
+		return S3SnapStoreModTime(config)
```

Comment:
Rename this to `GetS3CredentialsLastModifiedTime(config)`
Similar changes needs to be done for other providers as well. `S3SnapStoreModTime` does not mean anything